dev/tools/svg2dart/lib/svg2dart.dart

```diff
@@ -0,0 +1,547 @@
+}
+
+/// Represents a transformation to apply on an SVG subtree.
+///
+/// This includes more transforms than the ones described by the SVG transform
+/// attribute, e.g opacity.
+class _Transform {
+
+  /// Constructs a new _Transform, default arguments create a no-op transform.
+  _Transform({Matrix3 transformMatrix, this.opacity = 1.0}) :
+      this.transformMatrix = transformMatrix ?? new Matrix3.identity();
+
+  final Matrix3 transformMatrix;
+  final double opacity;
+
+  _Transform applyTransform(_Transform transform) {
+    return new _Transform(
+        transformMatrix: transform.transformMatrix.multiplied(transformMatrix),
+        opacity: transform.opacity * opacity,
+    );
+  }
+}
+
+final RegExp _transformCommand = new RegExp(' *(translate|scale|rotate)\\(([^)]*)\\)');
+
+Matrix3 _parseSvgTransform(String transform){
+  final Iterable<Match> matches =_transformCommand.allMatches(transform).toList().reversed;
+  Matrix3 result = new Matrix3.identity();
+  for (Match m in matches) {
+    final String command = m.group(1);
+    final String params = m.group(2);
+    if (command == 'translate') {
+      result = _parseSvgTranslate(params).multiplied(result);
+      continue;
+    }
+    if (command == 'scale') {
+      result = _parseSvgScale(params).multiplied(result);
+      continue;
+    }
+    if (command == 'rotate') {
+      result = _parseSvgRotate(params).multiplied(result);
+      continue;
+    }
+    throw new Exception('unimplemented transform: $command');
+  }
+  return result;
+}
+
+Matrix3 _parseSvgTranslate(String paramsStr) {
+  final List<String> params = paramsStr.split(',');
+  assert(params.isNotEmpty);
+  assert(params.length <= 2);
+  final double x = double.parse(params[0]);
+  final double y = params.length < 2 ? 0 : double.parse(params[1]);
+  return _matrix(1.0, 0.0, 0.0, 1.0, x, y);
+}
+
+Matrix3 _parseSvgScale(String paramsStr) {
+  final List<String> params = paramsStr.split(',');
+  assert(params.isNotEmpty);
+  assert(params.length <= 2);
+  final double x = double.parse(params[0]);
+  final double y = params.length < 2 ? 0 : double.parse(params[1]);
+  return _matrix(x, 0.0, 0.0, y, 0.0, 0.0);
+}
+
+Matrix3 _parseSvgRotate(String paramsStr) {
+  final List<String> params = paramsStr.split(',');
+  assert(params.length == 1);
+  final double a = radians(double.parse(params[0]));
+  return _matrix(cos(a), sin(a), -sin(a), cos(a), 0.0, 0.0);
+}
+
+Matrix3 _matrix(double a, double b, double c, double d, double e, double f) {
+  return new Matrix3(a, b, 0.0, c, d, 0.0, e, f, 1.0);
+}
+
+// Matches a pixels expression e.g "14px".
+// First group is just the number.
+final RegExp _pixelsExp = new RegExp('^([0-9]+)px\$');
+
+/// Parses a pixel expression, e.g "14px", and returns the number.
+/// Throws an [ArgumentError] if the given string doesn't match the pattern.
+int parsePixels(String pixels) {
+  if (!_pixelsExp.hasMatch(pixels))
+    throw new ArgumentError('illegal pixels expression: \'$pixels\'');
```

Comment:
it's probably not illegal, just not supported by this script

Reply:
Well... in the context of this function "parsePixels" it is illegal :) added a note to the exception